### PR TITLE
Creating Array Data Types UX improvements 

### PR DESCRIPTION
**Pull request by luisbettencourt**
## 🌟 What is the purpose of this PR?

This PR improves the UX when creating array data types.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203312852763953/1203481015428204/f) _(internal)_

## 🔍 What does this change?

- Prevent popper from open to the side;
- Trigger a window resize event whenever a new data type is added/removed so that the popper component recalculates its position;
- Add expand/collapse animations to data types when adding/removing them from the schema;
- Add expand/collapse width animation to the prefix ("CONTAINING", "OR", etc...) container.

## 📹 Demo

https://www.loom.com/share/efa8cb19fc414afb9a0e36efed612c66

### CODE REVIEW

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/data-type-badge.tsx [hunk 1]
```diff
@@ -47,9 +55,12 @@ export const DataTypeBadge = ({
             padding: 1.25,
             alignItems: "center",
             ...(isArray ? { borderBottomLeftRadius: 0 } : {}),
+            width: prefixContainerWidth,
+            transition: ({ transitions }) => transitions.create("width"),
```
Comment (benwerner01) [inline packages/hash/frontend/src/pages/[account-slug]/types/entity-type/data-type-badge.tsx:59]: I think this transition means it's possible to get into an odd state where the chip is jumping from line to line on hover. I think some right padding that is minimised when the width is expanded could address this.

https://user-images.githubusercontent.com/42802102/205361161-97d382e4-78c6-4082-a1e4-8ee89ae2440b.mov

**Review by benwerner01 — COMMENTED** [review]
The changes look good to me, spotted one minor visual bug that can occur in an edge-case.

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/data-type-badge.tsx (hunk 1 above)
Comment (benwerner01) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/data-type-badge.tsx:59]: This could also be addressed in follow-up.

Comment (luisbettencourt) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/data-type-badge.tsx:59]: Will do this in a follow-up

**Review by benwerner01 — APPROVED** [review]

**Pull request merged**